components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -16238,6 +16244,87 @@ public Set<String> getAPIVersions(String apiName, String apiProvider, String org
+                APIIdentifier identifier = new APIIdentifier(apiProvider, apiName, version);
+                api.setId(identifier);
+                api.setUuid(uuid);
+                api.setStatus(status);
+                api.setVersionTimestamp(versionTimestamp);
+                api.setContext(context);
+                api.setContextTemplate(contextTemplate);
+            } else {
+                handleException("Couldn't find API for given api name: " + apiName + ", api version: " + apiVersion);
+            }
+        } catch (SQLException e) {
+            handleException("Error while retrieving API info for api name: " + apiName + ", api version:" + apiVersion + " and provider: " + apiProvider,
+                    e);
+        }
+        return api;
+    }
+
+    /**
+     * Return ids of the versions for the given name for the given provider for an organization
+     *
+     * @param apiName     api name
+     * @param apiProvider provider
+     * @param organization organization name
+     * @return set ids
+     * @throws APIManagementException
+     */
+    public List<API> getAllAPIVersions(String apiName, String apiProvider, String organization) throws APIManagementException {
+
+        List<API> apiVersions = new ArrayList<API>();
+
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(SQLConstants.GET_API_VERSIONS_UUID_BY_ORGANIZATION)) {
+            statement.setString(1, APIUtil.replaceEmailDomainBack(apiProvider));
+            statement.setString(2, apiName);
+            statement.setString(3, organization);
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                String version = resultSet.getString("API_VERSION");
+                String status = resultSet.getString("STATUS");
+                String versionTimestamp = resultSet.getString("VERSION_COMPARABLE");
+                String context = resultSet.getString("CONTEXT");
+                String contextTemplate = resultSet.getString("CONTEXT_TEMPLATE");
+
+                String uuid = resultSet.getString("API_UUID");
+                if (APIConstants.API_PRODUCT.equals(resultSet.getString("API_TYPE"))) {
+                    // skip api products
+                    continue;
+                }
+                API api = new API(new APIIdentifier(apiProvider, apiName,
+                        version, uuid));
+                api.setUuid(uuid);
+                api.setStatus(status);
+                api.setVersionTimestamp(versionTimestamp);
+                api.setContext(context);
+                api.setContextTemplate(contextTemplate);
+                apiVersions.add(api);
+            }
+        } catch (SQLException e) {
+            handleException("Error while retrieving versions for api " + apiName + " for the provider " + apiProvider,
+                    e);
```

Comment:
Just throw an APIManagementException. No need to pass to another method

Reply:
Fixed with: https://github.com/wso2/carbon-apimgt/pull/12107/commits/054a492392c305d70924d72ad346694a7c75508e